### PR TITLE
fix integration tests

### PR DESCRIPTION
**Pull request by alejandroacho**
We were facing an issue with the `sleep` function during the tests. We were using 300000 milliseconds, which  made the test fail because of "inaction" since the code was waiting for so long. 

Since these are integration tests (this implies that we don't test the code itself, we build it and interact with it through the api in a docker container), we could not use mocks, patchers or make use of the jest timing functions.

The solution is to introduce an env variable to the backend and if is in test mode, force the `sleep` function to always sleep for 1 second.

### CODE REVIEW

**Review by markettes — APPROVED** [review]
LGTM

**Review by amanzag — COMMENTED** [review]
Thanks for debugging this.

I think though that changing the behavior of sleep() in all cases will give us problems in the future, and we won't even remember why it's hardcoded to 1000 in tests.

Couldn't it be changed just in the place where it's giving us problems? By the way, where is the place where we wait for 300000 ms?

File: server-backend/src/utils.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export function sleep(millis: number, abortSignal?: AbortSignal): Promise<void> {
-    const timeoutPromise = new Promise<void>(r => setTimeout(r, millis));
+    const timeoutPromise = new Promise<void>(r => setTimeout(r, process.env.TEST_MODE ? 1000 : millis));
```
Comment (alejandroacho) [inline server-backend/src/utils.ts:2]:
```suggestion
    const safeTestingTime = 1000; // This allow us to correctly execute integration tests
    const timeoutPromise = new Promise<void>(r => setTimeout(r, process.env.IS_TESTING ? safeTestingTime : millis));
```

@amanzag Maybe something like this? or do you still prefer to tackle this edge case in the function implementation?

Comment (amanzag) [reply server-backend/src/utils.ts:2]:
```suggestion
    const safeTestingTime = Math.min(1000, millis); // This allow us to correctly execute integration tests
    const timeoutPromise = new Promise<void>(r => setTimeout(r, process.env.IS_TESTING ? safeTestingTime : millis));
```
how about something like this? We should allow to sleep for less than a second.

**Comment by amanzag** [comment]
> > Thanks for debugging this.
> > I think though that changing the behavior of sleep() in all cases will give us problems in the future, and we won't even remember why it's hardcoded to 1000 in tests.
> > Couldn't it be changed just in the place where it's giving us problems? By the way, where is the place where we wait for 300000 ms?
> 
> @amanzag I think this will eventually be forgotten, and if we ever use a long sleep time, we'll have to debug the tests again. Therefore, I think it makes more sense to address this problem directly in the function itself, rather than in its implementation. Perhaps we can change the name to make it more self-explanatory and avoid hardcoded numbers? In the swap-in retry, we have that number of milliseconds: https://github.com/40acres/40swap/blob/master/server-backend/src/SwapInRunner.ts#L74

Ok, let's do it like this, but at least we need a comment in the code explaining why it's there.

File: server-backend/src/utils.ts (hunk 1 above)
Comment (alejandroacho) [reply server-backend/src/utils.ts:2]: lgtm

File: server-backend/src/utils.ts [hunk 2]
```diff
@@ -1,5 +1,9 @@
 export function sleep(millis: number, abortSignal?: AbortSignal): Promise<void> {
-    const timeoutPromise = new Promise<void>(r => setTimeout(r, millis));
+    const safeTestingTime = Math.min(1000, millis); // This allow us to correctly execute integration tests
+    // We need to make sure that the timeout is at least 1 second, otherwise the test will fail
```
Comment (amanzag) [inline server-backend/src/utils.ts:3]:
```suggestion
    // We need to make sure that the timeout is at most 1 second, otherwise the integration tests will fail.
```

Comment (alejandroacho) [reply server-backend/src/utils.ts]: Done

**Review by amanzag — APPROVED** [review]
